src/main/python/SystemML/mlcontext.py

```diff
@@ -122,22 +121,49 @@ def toDF(self):
 class MLResults(object):
     """
     Wrapper around a Java ML Results object.
-
     Parameters
     ----------
     results: JavaObject
         A Java MLResults object as returned by calling `ml.execute()`.
-
     sc: SparkContext
         SparkContext
     """
     def __init__(self, results, sc):
         self._java_results = results
         self.sc = sc
+        try:
+            if MLResults.sqlContext is None:
+                MLResults.sqlContext = SQLContext(sc)
+        except AttributeError:
+            MLResults.sqlContext = SQLContext(sc)
 
     def __repr__(self):
         return "MLResults"
 
+    def getNumPyArray(self, *outputs):
+        """
+        Parameters
+        ----------
+        outputs: string, list of strings
+            Output variables as defined inside the DML script.
+        """
+        outs = [convertToNumpyArr(self.sc, self._java_results.getMatrix(out).asBinaryBlockMatrix().getMatrixBlock()) for out in outputs]
+        if len(outs) == 1:
+            return outs[0]
+        return outs
+    
+    def getDataFrame(self, *outputs):
```

Comment:
I would remove this and just call `toDF()` on the resulting `Matrix` (or `matrix` if we merge them) from `get()`.